### PR TITLE
update plugin

### PR DESCRIPTION
**Pull request by fabianmieller**
With this solution, we can return results based on parameters.

For example: 
http://localhost:8080/api/v1/translations?locale=de_DE -> will return  `/v1/translations/GET_locale=de_DEjson`

### CODE REVIEW

**Comment by fabianmieller** [comment]
Possibility to optionally add config.routes

File: index.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ function rewrite(apiPath, extensions, req, next) {
 	const url = req.url.split("?");
 	const b = `${url[0]}/${req.method}`;
 	const mapped = extensions.some(ext => {
-		const resourcePath = `${b}.${ext}`;
+		const resourcePath = url[1] ? `${b}_${url[1]}.${ext}` : `${b}.${ext}`;
```
Comment (marosivanco) [inline index.js:9]: IMHO, this is single-purpose extension that inserts into path the whole query string, which includes potentially unsafe characters: equal sign, percent, ampersand.
https://www.mtu.edu/umc/services/digital/writing/characters-avoid/

**Comment by marosivanco** [comment]
> 
> 
> With this solution, we can return results based on parameters.
> 
> For example:
> http://localhost:8080/api/v1/translations?locale=de_DE -> will return `/v1/translations/GET_locale=de_DEjson`

Just few thoughts...
There is more than one possible mapping of query params, that make sense:
- /v1/translations/GET.json  (ignoring the locale param, this the usual case)
- /v1/translations/optional/deeper/nesting/GET.**de_DE**.json
- /v1/translations/optional/deeper/nesting/**de_DE**/GET.json
- /v1/translations/**de_DE**/optional/deeper/nesting/GET.json

The situation gets even more complicated with more params. Query params are basically a set. So the two requests with the same params, but different params ordering should still map to the same result.

The problem is already solvable with current state:
```javascript
pluginOptions: {
    "serve-api-mocks": {
        base: "/api",
            routes: [
            {
                path: "/v1/translations",
                callback(req, res, next) {
                    const filePath = `.${req.baseUrl}/${req.query.locale}/GET.json`;
                    if (FS.existsSync(filePath)) {
                        const url = req.url.split("?");
                        req.url = `${url[0]}/${req.query.locale}`;
                        next();
                    } else {
                        console.log(`GET ${req.originalUrl} → []`);
                        res.send([]).end();
                    }
                },
            },
        ],
    },
},
```
It is slightly different, than the mapping of your change.

Anyway, if you have any ideas how to make query parameters mapping more user-friendly without limiting general usability (too much), please let me know.

**Comment by fabianmieller** [comment]
Ok, I will close this request. This solution works for me. :)

**Pull request closed without merging**